setup.py

```diff
@@ -2,16 +2,24 @@
 import codecs
 import os
 from setuptools import setup, find_packages
-try:
-    from pip.req import parse_requirements
-    from pip.download import PipSession
-except ImportError:
-    print('Cannot find pip.')
-    raise
-
-# Get requirements from the requirements.txt file.
-pip_requirements = parse_requirements("requirements.txt", session=PipSession())
-install_requires = [str(ir.req) for ir in pip_requirements]
+
+import pip
+if pip.__version__ < "10":
+    try:
+        from pip.req import parse_requirements
+        from pip.download import PipSession
+    except ImportError:
+        print('Cannot find pip.')
+        raise
+    # Get requirements from the requirements.txt file.
+    pip_requirements = parse_requirements("requirements.txt", session=PipSession())
+    install_requires = [str(ir.req) for ir in pip_requirements]
+else:
+    def parse_requirements(filename):
+        """ load requirements from a pip requirements file """
+        lineiter = (line.strip() for line in open(filename))
+        return [line for line in lineiter if line and not line.startswith("#")]
```

Comment:
If we have this utility, I think we should use it in all occasions, and throw away the use of pip's version. That's less differences and less trouble maintaining two versions.

Reply:
Could we have a bunch of tests for this new function?